x/accounts/internal/implementation/context.go

```diff
@@ -123,9 +124,38 @@ func Whoami(ctx context.Context) []byte {
 	return ctx.Value(contextKey{}).(contextValue).whoami
 }
 
-type headerService struct{ header.Service }
+type headerService struct{ hs header.Service }
 
 func (h headerService) GetHeaderInfo(ctx context.Context) header.Info {
-	originalContext := ctx.Value(contextKey{}).(contextValue).originalContext
-	return h.Service.GetHeaderInfo(originalContext)
+	return h.hs.GetHeaderInfo(getOriginalContext(ctx))
+}
+
+var _ gas.Service = (*gasService)(nil)
+
+type gasService struct {
+	gs gas.Service
+}
+
+func (g gasService) GetGasMeter(ctx context.Context) gas.Meter {
+	return g.gs.GetGasMeter(getOriginalContext(ctx))
+}
+
+func (g gasService) GetBlockGasMeter(ctx context.Context) gas.Meter {
+	return g.gs.GetBlockGasMeter(getOriginalContext(ctx))
+}
+
+func (g gasService) WithGasMeter(ctx context.Context, meter gas.Meter) context.Context {
+	v := ctx.Value(contextKey{}).(contextValue)
+	v.originalContext = g.gs.WithGasMeter(v.originalContext, meter)
```

Comment:
minor nit: it seems to imply that use of `originalContext` indicates that it is unmodified since it was originally set/provided to `contextValue`. Given that we're modifying it here (really making a new one), consider renaming `originalContext`?

Reply:
renamed: originalContext=>parentContext

Reply:
mucho better